library/std/src/process.rs

```diff
@@ -2296,6 +2296,15 @@ impl Child {
 /// }
 /// ```
 ///
+/// In its current implementation, this function will execute exit handlers registered with `atexit`
+/// as well as other platform-specific exit handlers (e.g. `fini` sections of ELF shared objects).
+/// This means that Rust requires that all exit handlers are safe to execute at any time. In
+/// particular, if an exit handler cleans up some state that might be concurrently accessed by other
+/// threads, it is required that the exit handler performs suitable synchronization with those
+/// threads. (The alternative to this requirement would be to not run exit handlers at all, which is
+/// considered undesirable. Note that returning from `main` also calls `exit`, so making `exit` an
+/// unsafe operation is not an option.)
```

Comment:
@comex [sketched](https://github.com/rust-lang/rust/issues/126600#issuecomment-2302668950) an approach a Rust library could take when wrapping a C library with a problematic destructor, to still uphold this requirement. I don't know what the conclusion was regarding whether this would actually work, and I am not sure if it's worth spelling that out in these docs -- this seems more like something to go into some book on Rust FFI advice.